### PR TITLE
Checkpoint NGFW "set_threat_protection"

### PR DESCRIPTION
**Pull request by jmcadams-r7**
## Proposed Changes

This PR adds the "Set Threat Protection" action to Check Point NGFW

### Testing

```
Connect: Connecting...
Starting new HTTPS connection (1): 10.4.24.63:443
/usr/local/lib/python3.7/site-packages/urllib3-1.25.3-py3.7.egg/urllib3/connectionpool.py:851: InsecureRequestWarning: Unverified HTTPS request is being made. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.io/en/latest/advanced-usage.html#ssl-warnings
  InsecureRequestWarning)
https://10.4.24.63:443 "POST /web_api/login HTTP/1.1" 200 None
rapid7/Check Point NGFW:1.1.0. Step name: set_threat_protection
Starting new HTTPS connection (1): 10.4.24.63:443
/usr/local/lib/python3.7/site-packages/urllib3-1.25.3-py3.7.egg/urllib3/connectionpool.py:851: InsecureRequestWarning: Unverified HTTPS request is being made. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.io/en/latest/advanced-usage.html#ssl-warnings
  InsecureRequestWarning)
https://10.4.24.63:443 "POST /web_api/set-threat-protection HTTP/1.1" 200 None
Starting new HTTPS connection (1): 10.4.24.63:443
/usr/local/lib/python3.7/site-packages/urllib3-1.25.3-py3.7.egg/urllib3/connectionpool.py:851: InsecureRequestWarning: Unverified HTTPS request is being made. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.io/en/latest/advanced-usage.html#ssl-warnings
  InsecureRequestWarning)
https://10.4.24.63:443 "POST /web_api/publish HTTP/1.1" 200 None
Starting new HTTPS connection (1): 10.4.24.63:443
/usr/local/lib/python3.7/site-packages/urllib3-1.25.3-py3.7.egg/urllib3/connectionpool.py:851: InsecureRequestWarning: Unverified HTTPS request is being made. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.io/en/latest/advanced-usage.html#ssl-warnings
  InsecureRequestWarning)
https://10.4.24.63:443 "POST /web_api/logout HTTP/1.1" 200 None
{"body": {"log": "Connect: Connecting...\nrapid7/Check Point NGFW:1.1.0. Step name: set_threat_protection\n", "status": "ok", "meta": {}, "output": {"success": true}}, "version": "v1", "type": "action_event"}
```

## Validation: 
```
rmt-mbp-5357:checkpoint_ngfw jmcadams$ icon-validate .
[*] Validating plugin at .

[*] Running Integration Validators...
[*] Executing validator HelpValidator
[*] Executing validator ChangelogValidator
[*] Executing validator RequiredKeysValidator
[*] Executing validator UseCaseValidator
[*] Executing validator SpecPropertiesValidator
[*] Executing validator SpecVersionValidator
[*] Executing validator FilesValidator
[*] Executing validator TagValidator
[*] Executing validator DescriptionValidator
[*] Executing validator TitleValidator
[*] Executing validator VendorValidator
[*] Executing validator DefaultValueValidator
[*] Executing validator IconValidator
[*] Executing validator RequiredValidator
[*] Executing validator VersionValidator
[*] Executing validator DockerfileParentValidator
[*] Executing validator LoggingValidator
[*] Executing validator ProfanityValidator
[*] Executing validator AcronymValidator
[*] Executing validator JSONValidator
[*] Executing validator OutputValidator
[*] Executing validator RegenerationValidator
[*] Executing validator HelpInputOutputValidator
[*] Executing validator SupportValidator
[*] Plugin successfully validated!

----
[*] Total time elapsed: 333.469ms
```

## Workflow
image

### Artifacts
image

### Input
image

### Output
image

### CODE REVIEW

File: checkpoint_ngfw/plugin.spec.yaml [hunk 1]
```diff
@@ -692,6 +692,50 @@ actions:
         type: boolean
         required: true
 
+  set_threat_protection:
+    title: Set Threat Protection
+    description: Set Threat Protection
```
Comment (lmilby-r7) [inline checkpoint_ngfw/plugin.spec.yaml]:
```suggestion
    description: Set threat protection
```

File: checkpoint_ngfw/unit_test/test_set_threat_protection.py [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+from unittest import TestCase
+from icon_checkpoint_ngfw.connection.connection import Connection
+from icon_checkpoint_ngfw.actions.set_threat_protection import SetThreatProtection
+import json
+import logging
+
+
+class TestSetThreatProtection(TestCase):
+    def test_integration_set_threat_protection(self):
+        """
```
Comment (tylerslijboom-r7) [inline checkpoint_ngfw/unit_test/test_set_threat_protection.py]: I think these TODOs should be removed, you did them.

```suggestion
        """
```

File: checkpoint_ngfw/plugin.spec.yaml [hunk 3]
```diff
@@ -692,6 +692,50 @@ actions:
         type: boolean
         required: true
 
+  set_threat_protection:
+    title: Set Threat Protection
+    description: Set threat protection
```
Comment (tylerslijboom-r7) [inline checkpoint_ngfw/plugin.spec.yaml]: In light of some of the comments in the slack chat, Can you please expand on the description fields?  They are copies of the name field.

File: checkpoint_ngfw/plugin.spec.yaml [hunk 4]
```diff
@@ -692,6 +692,50 @@ actions:
...
+    input:
+      name:
+        title: Name
+        description: Name
+        required: true
+        type: string
+        order: 1
+      action:
+        title: Action
+        description: Action
```
Comment (jschipp-r7) [inline checkpoint_ngfw/plugin.spec.yaml]: Can we get a better description and with an example? `.... E.g. <here>`

File: checkpoint_ngfw/plugin.spec.yaml [hunk 5]
```diff
@@ -692,6 +692,50 @@ actions:
         type: boolean
         required: true
 
+  set_threat_protection:
+    title: Set Threat Protection
+    description: Set threat protection
+    input:
+      name:
+        title: Name
+        description: Name
```
Comment (jschipp-r7) [inline checkpoint_ngfw/plugin.spec.yaml]: Can we get a better description and with an example? .... E.g. <here>

File: checkpoint_ngfw/plugin.spec.yaml [hunk 6]
```diff
@@ -692,6 +692,50 @@ actions:
...
+        order: 2
+        enum:
+          - Inactive
+          - Detect
+          - Prevent
+          - Drop
+          - Accept
+      profile:
+        title: Profile
+        description: Profile
```
Comment (jschipp-r7) [inline checkpoint_ngfw/plugin.spec.yaml]: Can we get a better description and with an example? .... E.g. <here>

File: checkpoint_ngfw/plugin.spec.yaml (hunk 3 above)
Comment (jschipp-r7) [inline checkpoint_ngfw/plugin.spec.yaml]: Anythign we can put here that describe what threat protection does? Same in help.md

File: checkpoint_ngfw/plugin.spec.yaml (hunk 4 above)
Comment (jmcadams-r7) [reply checkpoint_ngfw/plugin.spec.yaml]: It's an ENUM

File: checkpoint_ngfw/help.md [hunk 7]
```diff
@@ -32,6 +32,33 @@ The connection configuration accepts the following parameters:
...
+
+##### Input
+
+|Name|Type|Default|Required|Description|Enum|
+|----|----|-------|--------|-----------|----|
+|action|string|None|True|Action|['Inactive', 'Detect', 'Prevent', 'Drop', 'Accept']|
+|discard_other_sessions|boolean|True|True|Discard all other user sessions. This can fix errors when objects are locked by other sessions|None|
+|name|string|None|True|Name|None|
+|profile|string|Optimized|True|Profile|None|
+
```
Comment (jschipp-r7) [inline checkpoint_ngfw/help.md]: Add Example input section

Example input:

```
{
  "action": "Inactive",
  "discard_other_sessions": true,
  "name": "Alt-N Technologies SecurityGateway Username Buffer Overflow",
  "profile": "Optimized"
}
```

File: checkpoint_ngfw/help.md [hunk 8]
```diff
@@ -32,6 +32,33 @@ The connection configuration accepts the following parameters:
 
 ### Actions
 
+#### Set Threat Protection
+
+This action is used to set Threat Protection.
```
Comment (mrinehart-r7) [inline checkpoint_ngfw/help.md]:
```suggestion
This action is used to set threat protection.
```

File: checkpoint_ngfw/plugin.spec.yaml [hunk 9]
```diff
@@ -692,6 +692,50 @@ actions:
...
 
+  set_threat_protection:
+    title: Set Threat Protection
+    description: Set threat protection action per profile
+    input:
+      name:
+        title: Name
+        description: Name of the protection to act on. e.g. Blaster Attacks
+        required: true
+        type: string
```
Comment (jschipp-r7) [inline checkpoint_ngfw/plugin.spec.yaml]: We can start adding the `example:` idea for each input. Moving forward with this

File: checkpoint_ngfw/plugin.spec.yaml [hunk 10]
```diff
@@ -457,18 +457,21 @@ connection:
...
     description: Server port
     required: true
+    example: 443
     default: 443
   ssl_verify:
     type: boolean
     title: SSL Verify
     description: Use SSL verification
     required: true
+    example: True
```
Comment (jschipp-r7) [inline checkpoint_ngfw/plugin.spec.yaml]:
```suggestion
    example: true
```

**Pull request merged**